src/cpp/session/modules/rmarkdown/SessionExecuteChunkOperation.hpp

```diff
@@ -0,0 +1,308 @@
+/*
+ * SessionExecuteChunkOperation.hpp
+ *
+ * Copyright (C) 2009-16 by RStudio, Inc.
+ *
+ * Unless you have received this program directly from RStudio pursuant
+ * to the terms of a commercial license agreement with RStudio, then
+ * this program is licensed to you under the terms of version 3 of the
+ * GNU Affero General Public License. This program is distributed WITHOUT
+ * ANY EXPRESS OR IMPLIED WARRANTY, INCLUDING THOSE OF NON-INFRINGEMENT,
+ * MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE. Please refer to the
+ * AGPL (http://www.gnu.org/licenses/agpl-3.0.txt) for more details.
+ *
+ */
+#ifndef SESSION_MODULES_RMARKDOWN_SESSION_EXECUTE_CHUNK_OPERATIONR_HPP
+#define SESSION_MODULES_RMARKDOWN_SESSION_EXECUTE_CHUNK_OPERATIONR_HPP
+
+#include <boost/bind.hpp>
+#include <boost/enable_shared_from_this.hpp>
+#include <boost/noncopyable.hpp>
+#include <boost/shared_ptr.hpp>
+
+#include <core/FileSerializer.hpp>
+#include <core/text/CsvParser.hpp>
+#include <core/system/Process.hpp>
+#include <core/system/ShellUtils.hpp>
+#include <core/system/System.hpp>
+
+#include "NotebookOutput.hpp"
+#include "NotebookExec.hpp"
+#include "SessionRmdNotebook.hpp"
+
+namespace rstudio {
+namespace session {
+namespace modules {
+namespace rmarkdown {
+namespace notebook {
+
+core::shell_utils::ShellCommand shellCommandForEngine(
+      const std::string& engine,
+      const core::FilePath& scriptPath)
+{
+   core::shell_utils::ShellCommand command(engine);
+   
+   if (engine == "Rscript")
+      command << "-f";
+   
+   command << scriptPath;
```

Comment:
Yes utf8ToSystem is what we want here (I wouldn't make any changes in ShellUtils but rather just add the conversion at the call site).